api/models/DetaineeEvent.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
+
+var LinkingModels = require('sails-linking-models');
+var ModelHelpers = require('../lib/ModelHelpers');
+
+const model = {
+  schema: true,
+  autoCreatedAt: true,
+  autoUpdatedAt: true,
+  attributes: {
+    id: {
+      type: 'string',
+      required: true,
+      defaultsTo: ModelHelpers.getPid
+    },
+    operation: {
+      type: 'string',
+      required: true,
+      enum: ['check in', 'update']
+    },
+    cid_id: {
+      type: 'integer',
+      required: true
+    },
+    gender: {
+      type: 'string',
+      enum: ['male', 'female'],
+      required: true
+    },
+    nationality: {
+      type: 'string'
+    },
+    centre: {
```

Comment:
We concluded that the centre didn't necessarily need to store anything, that counts etc, could be calculated on the fly, eg. The DetaineeEvent and Movement entities store data, which can be used in the scheduled count calculation when the request is made/DetaineeEvent table is updated.